### DISCUSSION TITLE
Add ability for SpotlightResult->when(...)

### DISCUSSION THREAD

**Discussion by nick-wilkins** [wire-elements/spotlight#118] [Ideas]
Some of my SpotlightResult::make commands I only want to register if the user is logged in. At the moment I am having to add an Auth::check() inside the SpotlightQuery::forMode (or token / default), and create 2 separate collections (some with repeated results).

I can refactor slightly of course to create common objects then create the collection afterwards, but it would be nice to have an:
->when(...) option, ie:
SpotlightResult::make()
   ->when(Auth::check())
    ->setTitle('Documentation')
    ...
    
If this already exists, or some form of, apologies but I can't see it (using Pro FYI)

**Comment by PhiloNL** [comment]
Hey @nick-wilkins,

The Collection has a when method so you could do something like:
```php
Spotlight::registerQueries(
  SpotlightQuery::asDefault(function ($query) {
    $user = request()->user();
    $pages = collect();
    $pages->when($user, fn($c) => $c->push(SpotlightResult::make(...)));
  });
```
Would that work?
I could add a `when` method to the result as well as you suggest; that might be easier and cleaner compared to having to use the collection when closure.
